File: natas11/natas11.php
<?php


	function xor_encrypt($in) {
	    $key = base64_decode("ClVLIh4ASCsCBE8lAxMacFMZV2hdVVotEhhUJQNVAmhSEV4sFxFeaAw=");
	    $text = $in;
	    $outText = '';

	    // Iterate through each character
	    for($i=0;$i<strlen($text);$i++) {
	    	$outText .= $text[$i] ^ $key[$i % strlen($key)];
	    }

	    return $outText;
	}

	$defaultdata = array( "showpassword"=>"no", "bgcolor"=>"#ffffff");

	$res = xor_encrypt(json_encode($defaultdata));

	print("Key Pattern : ".$res);

	
	//Generate new cookie

	$key = 'qw8J';

	$data = json_encode(array( "showpassword"=>"yes", "bgcolor"=>"#ffffff"));

	$outText = '';

    // Iterate through each character
    for($i=0;$i<strlen($data);$i++) {
    	$outText .= $data[$i] ^ $key[$i % strlen($key)];
    }

    $output = base64_encode($outText);

    print("\tNew cookie : ". $output);
?>

File: natas11/natas11.py
#!/usr/bin/python3

import base64

a = base64.b64decode("ClVLIh4ASCsCBE8lAxMacFMZV2hdVVotEhhUJQNVAmhSEV4sFxFeaAw=")
a = a.decode("ASCII")

b = '{"showpassword":"no","bgcolor":"#ffffff"}'
out = []

for i in range(len(a)):
	out.append(chr(ord(a[i])^ord(b[i%len(b)])))

out = ''.join(out)

print("key = ",out)

key = "qw8J"

data = '{"showpassword":"yes","bgcolor":"#ffffff"}'

out = []

for i in range(len(data)):
	out.append(chr(ord(data[i])^ord(key[i%len(key)])))

out = ''.join(out)

print("New Cookie = ", base64.b64encode(out))

